### PR TITLE
DDCNL-2834 | Refactor connector tests to use Wiremock

### PR DESCRIPTION
**Pull request by jpatton638**
- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: test/uk/gov/hmrc/tai/connectors/CitizenDetailsConnectorSpec.scala [hunk 1]
```diff
@@ -213,23 +158,23 @@ class CitizenDetailsConnectorSpec extends ConnectorBaseSpec with ScalaFutures wi
         None,
         "LName",
         "FName LName",
-        manualCorrespondenceInd = true,
-        Some(false))
+        manualCorrespondenceInd = false,
+        Some(true))
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply test/uk/gov/hmrc/tai/connectors/CitizenDetailsConnectorSpec.scala:162]: Original test was incorrect

File: test/uk/gov/hmrc/tai/connectors/ConnectorBaseSpec.scala [hunk 2]
```diff
@@ -15,18 +15,33 @@
...
-trait ConnectorBaseSpec extends PlaySpec with MockitoSugar with WireMockHelper {
+trait ConnectorBaseSpec extends PlaySpec with MockitoSugar with WireMockHelper with Injecting {
 
   val nino: Nino = new Generator(new Random).nextNino
+
   implicit val hc: HeaderCarrier = HeaderCarrier()
-  implicit lazy val cc: ExecutionContext = injector.instanceOf[ExecutionContext]
+  implicit lazy val ec: ExecutionContext = inject[ExecutionContext]
+
+  def assertConnectorException[A <: HttpException](call: Future[_], code: Int, message: String)(
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply test/uk/gov/hmrc/tai/connectors/ConnectorBaseSpec.scala:39]: Cheers!

File: test/uk/gov/hmrc/tai/connectors/NpsConnectorSpec.scala [hunk 3]
```diff
@@ -16,201 +16,836 @@
...
+            get(urlEqualTo(employmentsUrl)).willReturn(
+              aResponse()
+                .withStatus(OK)
+                .withBody(employmentListJson.toString())
+                .withHeader("ETag", s"$etag"))
+          )
 
-        val sut = createSUT(mockMetrics, mockHttpClient, mock[Auditor], mock[IabdUpdateAmountFormats], mock[NpsConfig])
-        Await.result(sut.getEmployments(nino, 2017)(HeaderCarrier()), Duration.Inf) mustBe expectedResult
+          Await.result(sut.getEmploymentDetails(nino, year), 5.seconds) mustBe employmentListJson
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply test/uk/gov/hmrc/tai/connectors/NpsConnectorSpec.scala:220]: There was no test for returning an actual payload, so I added it in

File: test/uk/gov/hmrc/tai/connectors/NpsConnectorSpec.scala [hunk 4]
```diff
@@ -16,201 +16,836 @@
...
+          server.stubFor(
+            get(urlEqualTo(iabdsUrl)).willReturn(
+              aResponse()
+                .withStatus(OK)
+                .withBody(body)
+                .withHeader("ETag", s"$etag"))
+          )
+
+          Await.result(sut.getIabds(nino, year), 5.seconds) mustBe
+            expectedResponse
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply test/uk/gov/hmrc/tai/connectors/NpsConnectorSpec.scala:339]: Again, there was no test for returning a correct payload, so I thought it was better to put this in

File: test/uk/gov/hmrc/tai/connectors/NpsConnectorSpec.scala [hunk 5]
```diff
@@ -16,201 +16,836 @@
...
+          server.stubFor(
+            get(urlEqualTo(iabdsForTypeUrl)).willReturn(
+              aResponse()
+                .withStatus(OK)
+                .withBody(body)
+                .withHeader("ETag", s"$etag"))
+          )
+
+          Await.result(sut.getIabdsForType(nino, year, iabdType), 5.seconds) mustBe
+            expectedResponse
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply test/uk/gov/hmrc/tai/connectors/NpsConnectorSpec.scala:458]: See above

File: test/uk/gov/hmrc/tai/connectors/FileUploadConnectorSpec.scala [hunk 6]
```diff
@@ -16,245 +16,312 @@
...
+            .withStatus(CREATED)
+            .withBody(envelopeId)
+            .withHeader("Location", s"/file-routing/requests/$envelopeId")
+        )
+      )
 
       Await.result(sut.closeEnvelope(envelopeId), 5 seconds) mustBe envelopeId
     }
 
-    "call the file upload service routing request endpoint" in {
```
[the remark this answers is not in the document]
Comment (jpatton638) [reply test/uk/gov/hmrc/tai/connectors/FileUploadConnectorSpec.scala:189]: Yes, the test above covers everything that this test does, so I removed it

**Review by struong — APPROVED** [review]

**Pull request merged**